journal/src/main/scala/com/evolutiongaming/kafka/journal/ActionHeader.scala

```diff
@@ -37,15 +37,57 @@ object ActionHeader {
             payloadType <- (json \ "payloadType").validate[PayloadType.BinaryOrJson]
             metadata    <- metadata
           } yield {
-            Append(range, origin, version, payloadType, metadata)
+            Append(range, origin, version.getOrElse(Version.obsolete), payloadType, metadata)
+          }
+        }
+      }
+      OFormat(reads, format)
+    }
+    val deleteFormat = {
```

Comment:
Could you somehow avoid having hand-crafter formatters? I have doubts that I wanna see them in a long term

Reply:
yes, I do not like them too - we could provide default value and use formatter whish supports default values